Added the use of the automatic variable $*

mboisson opened the pull request:
To replace $@. In my opinion, it is more readable using $*.dat than $@. The additional information ".dat" makes it clear that this is the target that we intend to generate.

mikej888 commented on 05-patterns.md:
```diff
@@ -8,7 +8,7 @@ minutes: 0
 >
 > * Write Make pattern rules.
-> * Use the Make wild-card `%` in targets and dependencies.
+> * Use the Make wild-card `%` in targets and dependencies and `$*` in actions.
```
Add `$*` as a separate learning objective on its own line.

mikej888 commented on 05-patterns.md:
```diff
@@ -19,10 +19,10 @@ rule](reference.html#pattern-rule) which can be used to build any
 ~~~
 
-`%` is a Make [wild-card](reference.html#wild-card).
+`%` is a Make [wild-card](reference.html#wild-card), and `$*` is a special variable which gets replaced by the stem with which the rule matched.
```
Start the `$*` text as a new sentence.

mikej888 commented:
As this changes the running example, all subsequent lessons and code samples need updated to be consistent:
```
07-functions.md
code/samples/05-patterns/Makefile
code/samples/06-variables-challenge/Makefile
code/samples/06-variables/Makefile
code/samples/07-functions/Makefile
code/samples/08-conclusion-challenge/Makefile
solutions.md
```
reference.md also needs to be updated to add `$*` to the Automatic variables list and the automatic variable entry in the glossary.

mboisson commented:
I believe I made the changes to all files to be consistent. 
I also added `stem` to the `reference.md` and added `$*` to the list of automatic variables.

mikej888 commented:
Thanks :-)
